src/middleware.js

```diff
@@ -179,6 +198,37 @@ function addLinkToCategory(category, link) {
   })
 }
 
+// return a copy of the whiteList
+async function getWhitelist() {
+  var p = new Promise(function(resolve, reject) {
+    chrome.storage.sync.get(["whitelist"], function (data) {
+      var list = data['whitelist']
+      if (list === undefined) {
+        list = []
+      }
+      resolve(list);
+    })
+  });
+  return await p; 
+}
+
+// update the whiteList (domains: list)
+function updateWhitelist(domains){
+  whitelistObj = {}
+  whitelistObj["whitelist"] = domains
+  chrome.storage.sync.set(whitelistObj, function() {
+    if (debugMode) {
+      console.log('Update white list: ' + domains);
+      console.log(whitelistObj);
+    }
+  });
+}
+
+// update doTrack variable (option: boolean)
+function enableTracking(option) {
```

Comment:
maybe we make two functions, enableTracking and disableTracking to be more clear. This is a nit pick, not a big deal

Reply:
Yea, but this was written in the design document and I was just following the doc.